projects/packages/jetpack-mu-wpcom/tests/php/features/launchpad/class-launchpad-task-lists-test.php

```diff
@@ -22,4 +22,101 @@ public function test_build_creates_no_PHP_warnings() {
+		return $first_task['completed'];
+	}
+
+	/**
+	 * Data provider for ::test_task_list_is_completed().
+	 */
+	public function provider_test_task_list_is_completed() {
+		$incomplete_task = array();
+		$completed_task  = array(
+			'is_complete_callback' => '__return_true',
+		);
+
+		$launch_task_incomplete = array(
+			'isLaunchTask' => true,
+		);
+
+		$launch_task_completed = array(
+			'isLaunchTask'         => true,
+			'is_complete_callback' => '__return_true',
+		);
+
+		return array(
+			// tasks, expected, task_list_options
+			'all tasks incomplete should be incomplete using default options'
+				=> array( array( $incomplete_task, $incomplete_task ), false ),
+			'all tasks complete should be complete using default options'
+				=> array( array( $completed_task, $completed_task ), true ),
+			'incomplete task and incomplete launch task should be incomplete using default options'
+				=> array( array( $incomplete_task, $launch_task_incomplete ), false ),
+			'having the launch task completed should make the task list complete'
+				=> array( array( $incomplete_task, $launch_task_completed ), true ),
+			'with the require_last_task_completion option, the last task being complete should make the task list complete'
+				=> array( array( $incomplete_task, $completed_task ), true, array( 'require_last_task_completion' => true ) ),
+			'with the require_last_task_completion option, the last task being incomplete should make the task list incomplete'
+				=> array( array( $completed_task, $incomplete_task ), false, array( 'require_last_task_completion' => true ) ),
+			'with the required_task_ids option, the task list should be complete if the required task is complete'
+				=> array( array( $completed_task, $incomplete_task ), true, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			'with the required_task_ids option, the task list should be incomplete if the required task is incomplete'
+				=> array( array( $incomplete_task, $incomplete_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			'with the required_task_ids option and the last task incomplete, the task list should be incomplete if the required task is incomplete'
+				=> array( array( $incomplete_task, $completed_task ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			'the launch task completion should not mark the task list complete if the required task is incomplete'
+				=> array( array( $incomplete_task, $launch_task_completed ), false, array( 'required_task_ids' => array( 'task_0' ) ) ),
+			'overriding the is_completed_callback should be the only thing that matters'
+				=> array( array( $incomplete_task, $incomplete_task ), true, array( 'is_completed_callback' => '__return_true' ) ),
+			'custom is_completed_callback should be able to check if first task is complete'
+				=> array( array( $completed_task, $incomplete_task ), true, array( 'is_completed_callback' => array( $this, 'callback_first_task_completed' ) ) ),
```

Comment:
Question: can we use an inline function for the callback? I don't know if that actually helps with overall readability, but it does keep the logic closer to the test case.
```suggestion
				=> array(
		array( $completed_task, $incomplete_task ),
		true,
		array(
			'is_completed_callback' => function ( $task_list ) {
				$first_task = reset( wpcom_launchpad_checklists()->build( $task_list['id'] ) );
				return $first_task['completed'];
			},
		),
	),
```